### PR TITLE
update drishti pointer, etc

### PR DESCRIPTION
**Pull request by headupinclouds**
* update hunter
* use drishti submodule and update pointer
* add comment line focal length input
* add comments
* HUNTER_KEEP_PACKAGE_SOURCES=ON

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ addons:
   apt:
     sources:
       - ubuntu-toolchain-r-test
+      - llvm-toolchain-trusty-5.0
```
Comment (ruslo) [inline .travis.yml]: This configuration runs Clang 5.0.0 too:
* https://github.com/ingenue/hunter/blob/3834d1dff2af8424b65887b51f7ffdfab3870c01/.travis.yml#L12-L24

File: bin/hunter_env.sh [hunk 2]
```diff
@@ -39,7 +39,9 @@ git submodule update --init --recursive --quiet
 uname -a
 
 # Install Python 3
-if [[ "$(uname)" == "Darwin" ]] && [ ! $(which python3) ]; then travis_retry brew install python3; fi
+# >> Error: python 2.7.14 is already installed \n To upgrade to 3.6.4_3, run `brew upgrade python`
+# if [[ "$(uname)" == "Darwin" ]] && [ ! $(which python3) ]; then travis_retry brew install python3; fi
```
Comment (ruslo) [inline bin/hunter_env.sh]: See https://github.com/ingenue/hunter/blob/3834d1dff2af8424b65887b51f7ffdfab3870c01/.travis.yml#L104-L106

File: .travis.yml [hunk 3]
```diff
@@ -35,13 +36,27 @@ addons:
...
+      - libc++-dev
+      - libc++abi-dev
+      - libclang-5.0-dev
+      - libclang-common-5.0-dev
+      - libclang1-5.0
+
       # pip3
       - python3-pip
 
+env:
```
Comment (ruslo) [inline .travis.yml]: Problem with this token `HUNTER_RUN_UPLOAD: OFF`

File: bin/hunter_env.sh (hunk 2 above)
Comment (headupinclouds) [reply bin/hunter_env.sh]: i'll update it.  thanks.

File: .travis.yml (hunk 1 above)
Comment (headupinclouds) [reply .travis.yml]: okay, so if we update to gcc-7 toolchains we can get clang-5 without an additional installation step?

File: .travis.yml (hunk 3 above)
Comment (headupinclouds) [reply .travis.yml]: fixed ... finally!

File: .travis.yml (hunk 1 above)
Comment (ruslo) [reply .travis.yml]: It seems so. Path to Clang will be `/usr/local/clang-5.0.0/bin/clang` (set to PATH by Travis) and you can use `clang-cxx17` toolchain.

**Pull request merged**